contracts/Stake/StakeManager.sol

```diff
@@ -307,6 +308,46 @@ contract StakeManager is Initializable, Ownable {
         emit Redelegated(_stashId, _updatedCluster);
     }
 
+    function splitStash(bytes32 _stashId, bytes32[] calldata _tokens, uint256[] calldata _amounts) external {
+        Stash memory _stash = stashes[_stashId];
+        require(
+            _stash.staker == msg.sender, 
+            "SM:SS - Only staker of stash can split"
+        );
+        require(
+            _tokens.length != 0,
+            "SM:SS - split stash must have atleast one token"
+        );
+        require(
+            _tokens.length == _amounts.length,
+            "SM:SS - Token data invalid"
+        );
+        uint256 _stashIndex = stashIndex;
+        bytes32 _newStashId = keccak256(abi.encodePacked(_stashIndex));
+        for(uint256 _index=0; _index < _tokens.length; _index++) {
+            bytes32 _tokenId = _tokens[_index];
+            uint256 _amount = _amounts[_index];
+            require(
+                stashes[_newStashId].amount[_tokenId] == 0,
+                "SM:SS - Can't add the same token twice while splitting stash"
+            );
+            require(
+                _amount != 0,
+                "SM:SS - Can't add tokens with 0 amount"
+            );
+            stashes[_stashId].amount[_tokenId] = stashes[_stashId].amount[_tokenId].sub(
+                _amount, 
+                "SM:SS-Insufficient Balance"
+            );
+            stashes[_newStashId].amount[_tokenId] = _amount;
+        }
+        stashes[_newStashId].staker = msg.sender;
+        stashes[_newStashId].delegatedCluster = _stash.delegatedCluster;
+        stashes[_newStashId].undelegatesAt = _stash.undelegatesAt;
+        emit StashSplit(_newStashId, _stashId, _stashIndex, _tokens, _amounts);
+        stashIndex = _stashIndex + 1;
```

Comment:
refactor this out into an internal `_createStash` function that's used by create stash as well

Reply:
right, will do.